### ISSUE TITLE
[Docs] eager backward snippet in getting-started/core-concepts.md does not compile

### ISSUE THREAD

**Issue by terasakisatoshi** [tensor4all/tenferro-rs#1116]
## Problem

The "Eager Execution And Backward" snippet in `docs/getting-started/core-concepts.md` uses `(&x * &x).reduce_sum(&[0]).unwrap()`, but `EagerTensor` does not implement `Mul` for references. Copy-pasting the snippet into a fresh crate produces a compiler error.

## Reproducer

Create a new crate with:

```toml
[dependencies]
tenferro-ad = { path = "/path/to/tenferro-rs/crates/tenferro-ad" }
```

Then paste the snippet from `docs/getting-started/core-concepts.md`:

```rust
use tenferro_ad::{EagerRuntime, Tensor};

let ctx = EagerRuntime::new();
let x = ctx.variable_from(Tensor::from_vec_col_major(vec![2], vec![1.0_f64, 2.0]));
let loss = (&x * &x).reduce_sum(&[0]).unwrap();
loss.backward().unwrap();

assert_eq!(x.grad().unwrap().as_slice::<f64>().unwrap(), &[2.0, 4.0]);
```

Compiler error:

```text
error[E0369]: cannot multiply `&EagerTensor` by `&EagerTensor`
```

## Expected behavior

The getting-started snippet should compile and run as documented, producing gradients `[2.0, 4.0]`.

## Actual behavior

The snippet fails to compile because `EagerTensor` binary operations are fallible methods (`add`, `mul`, etc.) rather than operator overloads.

## Affected area

Docs / `EagerTensor` / `tenferro-ad`

## Proposed fix

Update the snippet to use the method form and explain that the operations return `Result`:

```rust
use tenferro_ad::{EagerRuntime, Tensor};

let ctx = EagerRuntime::new();
let x = ctx.variable_from(Tensor::from_vec_col_major(vec![2], vec![1.0_f64, 2.0]));
let loss = x.mul(&x).unwrap().reduce_sum(&[0]).unwrap();
loss.backward().unwrap();

assert_eq!(x.grad().unwrap().as_slice::<f64>().unwrap(), &[2.0, 4.0]);
```

## Acceptance criteria

- [ ] `docs/getting-started/core-concepts.md` eager backward snippet compiles when copied into a fresh crate.
- [ ] The documented output `[2.0, 4.0]` is produced.

**Comment by shinaoka** [comment]
Fixed by merged PR #1125. The getting-started eager backward snippet and tutorial code were updated and verified by the docs checks recorded in the PR work log.
